osu.Game/Rulesets/UI/RulesetInputManager.cs

```diff
@@ -38,6 +40,14 @@ protected override InputState CreateInitialState()
 
         private readonly Container content;
 
+        public event Func<T?, List<T>, bool> HandleBindings
+        {
+            add => ((RulesetKeyBindingContainer)KeyBindingContainer).HandleBindings += value;
+            remove => ((RulesetKeyBindingContainer)KeyBindingContainer).HandleBindings -= value;
+        }
+
+        public void RetractLastBlocked() => ((RulesetKeyBindingContainer)KeyBindingContainer).RetractLastBlocked();
```

Comment:
These changes should not reside here publicly if at all avoidable. You should try to extract a ruleset input manager implementation specific to your mod that contains the blocking behaviour and expose it there only.